### PR TITLE
machinst x64: enable CLIF testing

### PR DESCRIPTION
**Pull request by bnjbvr**
This adds a new feature experimental_x64 for CLIF tests.

A test is run in the new x64 backend iff:

- either the test doesn't have an x86_64 target requirement, signaling
it must be target agnostic or not run on this target.
- or the test does require the x86_64 target, and the test is marked
with the `experimental_x64` feature.

This required one workaround in the parser. The reason is that the
parser will try to use information not provided by the TargetIsa adapter
for the Mach backends, like register names. In particular, parsing test
may fail before the test runner realizes that the test must not be run.
In this case, we early return an almost-empty TestFile from the parser,
under the same conditions as above, so that the caller may filter out
the test properly.

This also copies two tests from the test suite using the new backend,
for demonstration purposes.

### CODE REVIEW

File: cranelift/filetests/filetests/vcode/x64/basic.clif [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    ; check: pushq   %rbp
+    ; check: movq    %rsp, %rbp
+    v2 = iadd v0, v1
+    ; check: addl    %esi, %edi
+    return v2
+    ; check: movq    %rdi, %rax
+    ; check: movq    %rbp, %rsp
+    ; check: popq    %rbp
+    ; check: ret
+}
```
Comment (abrown) [inline cranelift/filetests/filetests/vcode/x64/basic.clif:16]: I would have expected these tests under `.../filetests/isa/x64`?

Comment (abrown) [reply cranelift/filetests/filetests/vcode/x64/basic.clif:16]: And looking at the `filetest` structure more closely, what do you think about a slight re-organization in a separate PR:
 - `filetests/vcode/*` get moved under `filetests/isa`
 - move a bunch of the optimization directories under a new `filetests/opt`
I'm just thinking that finding where to put new tests is getting challenging.

**Review by abrown — APPROVED** [review]
I really wish this was all more straightforward but this is probably the best it can be right now: at least now we can start duplication existing `isa/x86` tests to `x64` with the new feature flag and re-use some of that work.

File: cranelift/filetests/filetests/vcode/x64/basic.clif (hunk 1 above)
Comment (bnjbvr) [reply cranelift/filetests/filetests/vcode/x64/basic.clif:16]: I've moved them over to isa/x64; for the rest, I'm not sure what you mean by the optimization directories, so we could discuss it somewhere convenient.

**Pull request merged**